### ISSUE TITLE
Forwarding to current port

### ISSUE THREAD

**Issue by tfeige91** [expfactory/expfactory-docker#184]
Hi, 

I just tried running two sets of tests on the same server mapping to different ports: 

First: docker run -d -p 80:80 expfactory/experiments --database "DATABASE" --experiments code,simple-reaction-time,stroop,go-nogo,bart,n-back,tower-of-london,kirby,flanker,stop-signal,reversal-learning,judgement-of-line-orientation-test,information-sampling start

Second: docker run --name="impulsivity" -d -p 3391:80 expfactory/experiments --database "DATABASE" --experiments code,simple-reaction-time,flanker,go-nogo,kirby,stop-signal,bart,stroop,information-sampling,reversal-learning start

Basically it works good but if you start the tests the forwarding leads you to SERVER:80 not SERVER:3391. 
Is it possible to change that?

Thanks :)
Tim

**Comment by vsoch** [comment]
For both of those, you mapped to port 80 on the host. If you want to have tests on a different port, you need to map to it. The way it works is like `<port inside container>:<port on your host>`

**Comment by tfeige91** [comment]
Do I need to expose that Port in the dockerfile as well?
Should both of them not be mapped to 80?

**Comment by vsoch** [comment]
Yes that's correct! Any port that you use from inside the container needs:
```
EXPOSE <port>
EXPOSE 3391
```
But since each container is isolated, what you really should care about are the ports you map to on the host (the second in the mapping). Using port 80 inside each of the containers should work best unless you are doing some significant change to services in the container that require running the expfactory server on another port.

**Comment by tfeige91** [comment]
Hi Vanessa,
thanks for your help!

Unfortunately, mapping 80:3391 fails with _Bind for 0.0.0.0:80 failed: port is already allocated._

Maybe I'm doing it wrong: Do I need to actually use _docker build_ twice?

**Comment by vsoch** [comment]
That means you have something else running on port 80, perhaps another container? Or nginx / apache on your host? What is your goal here - to run two different experiment servers at once on the same host? if you want to do that, you need to map each container to a different _port on the host_, e.g.,

```
80:3001 # map the container's internal port 80 to 3001 on your host
80:3002 # map the container's internal port 80 to 3002 on your host
```

**Comment by tfeige91** [comment]
Hi Vanessa,
thanks to your help with my questions in autumn I managed to build a working system. 
This here however I didn't get (right now I stop one container and start the other one when another test-system is needed - that writes into another database).

What i would love to achieve is having one set of tests under URL/testsystem1 and the other one under URL/testsystem2 OR
URL:Port1 / URL:Port2.
On the first look it works: I can see the different test-systems in the different ports but once there is any change in the site (e.g. if you start the tests) you are forwarded to Port 80.

The port mappings you suggested above leed to the sites not being accessible.

I used the dockerfile from this issue-post: https://github.com/expfactory/expfactory/issues/125

Might this be due to the nginx-configuration in the container?

Thanks for your help and kind regards!
Tim

**Comment by vsoch** [comment]
The port needs to be exposed in the docker-compose.yml file too, and you likely need to tweak the `DOMAIN_NAME` in settings. Try both of those things (and sometimes you need to re-create the containers for it to work) and let me know if it works.

**Comment by tfeige91** [comment]
I exposed the ports, but I don't know where to change the DOMAIN_NAME. Where are the _settings_ ?

**Comment by vsoch** [comment]
Should be these two lines https://github.com/expfactory/expfactory-docker/blob/master/expdj/settings.py#L19. Whether you rebuild or just bind and restart, you can shell into the container to double check that they are changed.

**Comment by tfeige91** [comment]
Thanks for the super quick reply! Ah I see. By the way am i using the wrong version? This seems to be the one that has last been updated two years ago. 

I always use this to build the container: docker build --no-cache  -t expfactory/experiments .
Can I update this file from inside the container?

But this pulls from Dockerhub not quay.io , right?

**Comment by vsoch** [comment]
Oh sorry - are you using the expactory-docker code in the repository here, or version 2 of expfactory (at https://expfactory.github.io)? Sorry they are different! If you can clarify which one, I can better help (I thought you were using the one here since the issue is posted here).

**Comment by tfeige91** [comment]
Oh sorry, I use version 2 (the one at https://expfactory.github.io), although it says _version 3.1_ on the website. I didn't know there where differences by the time I created this issue.
